proposed/2023/release-staging-and-deprecation.md

```diff
@@ -0,0 +1,86 @@
+
+## Motivation 
+
+<!-- Why are we doing this? What pain points does this solve? What is the expected outcome? -->
+Many products (Apps / Tools / Frameworks / Utilities / etc. ...) consist of multiple interdependent packages which should be published together as a single set. Currently, NuGet provides no functionality for definding / grouping packages into such sets. This results in package owners having to keep track of these package sets on their own. There are three specific sets of scenarios which this feature will simplify dramatically:
+1. When preparing a release, product owners would like to have a mechanism to pre-stage their product packages, so that the release will consist only of a single action of marking the set of packages as `publishged`, which would result all the packages in the set to be marked as published at the same time.
+2. Sometimes developers have to take a dependency on several related packages in their projects, which are part of the same "release". Without having a concept of the `release` developers have to guess which versions of those packages are compatible. As a result, most of hte time developers would simply pick the latest version of all these packages, which somtimes can lead to incompatibility issues down the road.
+3. Keeping a package healthy on NuGet also means keeping the list of packages aligned with it support policy. Hence, package owners try to deprecate packages which are out of support. Without ability to group packages, package owners end up keeping track of all the packages which belong to the same release and run the deprecation over a discrete set of packages when necessary. Being able to group packages together would enable deprecating a whole release at once, knowing that all the packages which belong to that release will get deprecated.
+
+## Explanation
+
+### Functional explanation
+
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+There is a new field in the package metadata, which indicates the unique release of the product the package belongs to. This field is called `release` and the value is scoped to the publisher of the package.
+
+When a package is being built, that field in the metadata is set to the same value for all the related packages, that are built together.
+
+Now, with this information in place, new set of NuGet APIs / extensions allow the user to run the following actions:
```

Comment:
To be clear, is this proposal suggesting to change the [NuGet Server API](https://learn.microsoft.com/en-us/nuget/api/overview) and therefore all NuGet feeds can (should?) implement?  Or is the proposal for nuget.org only?

How much thought should we put into how much the feature makes sense in contexts other than nuget.org, such as [the many servers listed in our docs on hosting your own feeds](https://learn.microsoft.com/en-us/nuget/hosting-packages/overview). Some of the servers seem more tailored to hosting a single feed, similar to nuget.org, whereas others, particularly the SaaS offerings, seem to be more tailored towards an integrated CI/CD or DevOps environment

Reply:
Any API changes should be documented on the https://learn.microsoft.com/en-us/nuget/api/overview now, or perhaps eventually, instead of staying on a page like https://learn.microsoft.com/en-us/nuget/api/nuget-protocols forever. The NuGet team is steward to the NuGet ecosystem which intersects with concerns from other feed implementors. 

For the case of the deprecation API, NuGet.org is the first to implement it and it's in a private preview state only on NuGet.org, but when it graduates to public preview it will be documented on the NuGet Server API much like the push or unlist/delete API so other feeds can pick it up if they want.